### ISSUE TITLE
Kokkos:  `mdspan` include directory set incompletely

### ISSUE THREAD

**Issue by maartenarnst** [trilinos/Trilinos#12984]
We're trying to compile Kokkos and Trilinos using `mdspan` by setting the `CMake` variable `Kokkos_ENABLE_IMPL_MDSPAN` to `ON`.  

Enabling `mdspan` in this way works when we compile Kokkos as a standalone.

However, when we compile Kokkos as part of Trilinos, compilation fails with error messages like:

```
INFO:root:#20 127.4 /Trilinos-sources/packages/kokkos/core/src/View/MDSpan/Kokkos_MDSpan_Header.hpp:47:10: fatal error: mdspan/mdspan.hpp: No such file or directory
INFO:root:#20 127.4    47 | #include <mdspan/mdspan.hpp>
```

When we check the compile commands for the build with Kokkos as part of Trilinos, we see that the include directory `../../tpls/mdspan/include` is added only to the files in the `kokkos/core/src` directory.

The culprit might be that the use of `KOKKOS_LIB_INCLUDE_DIRECTORIES` does not append the include directory to the `CMake` target:

- https://github.com/kokkos/kokkos/blob/a75dc70d859b51b0ea8ccb697237e355e4f2f20a/core/src/CMakeLists.txt#L165-L168 
- https://github.com/kokkos/kokkos/blob/a75dc70d859b51b0ea8ccb697237e355e4f2f20a/cmake/kokkos_tribits.cmake#L486-L489

@romintomasetti @ndellingwood @masterleinad @crtrott

**Comment by jhux2** [comment]
@trilinos/kokkos

**Comment by ndellingwood** [comment]
@maartenarnst hopefully this now resolved in the 4.4 release, merged in PR #13334
